### PR TITLE
Networking Clean Up and Bug Fixes

### PR DESCRIPTION
**Pull request by ReadeP**
## Overview

Cleaned up some of the networking code and fixed a few bugs.

## Changes Made

- Refactored `NetworkingManager` and `ApiManager` to use `Result` type for cleaner code and more reliable error handling.
- Fixed bug where user's major was not appearing in `EditDemographicsViewController`.
- Fixed glitchy navigation bug in `HomeViewController` when user selected someone in the `CommunityViewController`.
- Various code fixes and improvements

## Test Coverage

Manually tested on device.

## Next Steps (delete if not applicable)

Continuing refactoring networking and remove outdated models.
Update the UI to allow for more than 1 major.

### CODE REVIEW

**Review by danielVebman — APPROVED** [review]
Great PR to kick off the semester! I left a couple comments here and there. 🤩🚀🚢

File: Pear/Controller/LoginViewController.swift [hunk 1]
```diff
@@ -101,14 +101,20 @@ class LoginViewController: UIViewController {
...
-                        animated: false
-                    )
+                switch result {
+                case .success(let userSession):
+                    DispatchQueue.main.async {
+                        UserDefaults.standard.set(userSession.accessToken, forKey: Constants.UserDefaults.accessToken)
+                        
+                        let onboardingVC = OnboardingPageViewController(transitionStyle: .scroll, navigationOrientation: .horizontal)
+                        
+                        self.navigationController?.pushViewController(onboardingVC, animated: false)
```
[the remark this answers is not in the document]
Comment (ReadeP) [reply Pear/Controller/LoginViewController.swift:114]: This is how it was before, but I think it looks better animated

File: Pear/Controller/TabPageViewController.swift [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 import UIKit
 
-protocol TabDelegate: class {
+protocol TabDelegate: AnyObject {
```
[the remark this answers is not in the document]
Comment (ReadeP) [reply Pear/Controller/TabPageViewController.swift:11]: It's declared as a weak variable which requires TabDelegate to conform to a class

`'weak' must not be applied to non-class-bound 'TabDelegate'; consider adding a protocol conformance that has a class bound`

File: Pear/Networking/NetworkManager.swift [hunk 3]
```diff
@@ -25,7 +25,7 @@ class NetworkManager {
 
     private static let hostEndpoint = "http://\(Keys.pearServerURLV2)"
 
-    static func uploadPhoto(base64: String, completion: @escaping (String) -> Void) {
+    static func uploadPhoto(base64: String, completion: @escaping (Result<String, Error>) -> ()) {
```
[the remark this answers is not in the document]
Comment (ReadeP) [reply Pear/Networking/NetworkManager.swift:28]: gross 🤮

File: Pear/Networking/NetworkManager.swift [hunk 4]
```diff
@@ -494,13 +510,14 @@ class NetworkManager {
...
-                    }
+                    let matchData = try jsonDecoder.decode(Response<MatchV2>.self, from: data)
+                    let match = matchData.data
+                    completion(.success(match))
+                } catch {
+                    completion(.failure(error))
                 }
             case .failure(let error):
-                print(error.localizedDescription)
+                completion(.failure(error))
```
[the remark this answers is not in the document]
Comment (ReadeP) [reply Pear/Networking/NetworkManager.swift:520]: Agreed

File: Pear/Onboarding/DemographicsViewController.swift [hunk 5]
```diff
@@ -141,26 +141,36 @@ class DemographicsViewController: UIViewController {
...
-                    self.classDropdownView.setTitle(title: graduationYear)
+            switch result {
+            case .success(let user):
+                DispatchQueue.main.async {
+                    if let graduationYear = user.graduationYear {
+                        self.classDropdownView.setTitle(title: graduationYear)
+                    }
+                    self.hometownDropdownView.setTitle(title: user.hometown ?? "")
+                    self.pronounsDropdownView.setTitle(title: user.pronouns ?? "")
+                    self.majorDropdownView.setTitle(title: user.majors.first?.name ?? "")
```
[the remark this answers is not in the document]
Comment (ReadeP) [reply Pear/Onboarding/DemographicsViewController.swift:170]: A change from our previous backend to the current one is users can have multiple majors. I don't think there is UI to reflect this however, so for now we only show the first major.

File: Pear/Profile/ProfileViewController.swift [hunk 6]
```diff
@@ -62,24 +62,28 @@ class ProfileViewController: UIViewController {
...
+            case .success(let user):
+                self.user = user
+                self.profileSections = [.summary, .basics]
+                if !user.interests.isEmpty {
+                    self.profileSections.append(.interests)
+                }
+                if !user.groups.isEmpty {
+                    self.profileSections.append(.groups)
+                }
+                self.profileTableView.reloadData()
```
[the remark this answers is not in the document]
Comment (ReadeP) [reply Pear/Profile/ProfileViewController.swift:83]: Yes it should be

**Pull request merged**